### PR TITLE
Add regression test for GH-1647

### LINKED ISSUE #1686: Test for pip-compile cached VCS packages
A tech-debt issue to address https://github.com/jazzband/pip-tools/pull/1649#discussion_r984986932.

### PR DESCRIPTION
**Pull request by gschaffner**
Closes #1686.

This asserts that the package is cached to avoid the problem with the prior test (where it passed when it should have failed—see https://github.com/jazzband/pip-tools/pull/1649#discussion_r939591090).

##### Contributor checklist

- [x] Provided the tests for the changes.
- [ ] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: tests/test_cli_compile.py [hunk 1]
```diff
@@ -511,6 +511,64 @@ def test_editable_package_vcs(runner):
...
+                "--format=abspath",
+                vcs_wheel_prefix,
+            ],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout
+    )
+
+    with open("requirements.in", "w") as req_in:
```
Comment (webknjaz) [inline tests/test_cli_compile.py:563]: Why is this written to the project root?

Comment (gschaffner) [reply tests/test_cli_compile.py:554]: Not sure. I just followed the pattern used already for the other tests.

**Comment by gschaffner** [comment]
Rebased onto `master`, (hopefully) fixed the Windows incompatibility, and force pushed.

**Comment by gschaffner** [comment]
Rebased onto `master`, (hopefully) fixed the Windows incompatibility, and force pushed.

**Comment by gschaffner** [comment]
I rebased onto `master` and fixed the Windows incompatibility.

**Review by atugushev — APPROVED** [review]
Thanks a lot!

**Pull request merged**